components/datepicker/DatePickerRange.js

```diff
@@ -0,0 +1,107 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import DayPicker from 'react-day-picker';
+import NavigationBar from './NavigationBar';
+import WeekDay from './WeekDay';
+import { convertModifiersToClassnames, hasRange, isSelectingFirstDay } from './utils';
+import cx from 'classnames';
+import theme from './theme.css';
+
+class DatePickerRange extends PureComponent {
+  state = {
+    from: null,
+    to: null,
+    enteredTo: null,
+  };
+
+  static getDerivedStateFromProps(props, state) {
+    if (
+      props.selectedRange !== undefined &&
+      (props.selectedRange.from !== state.from || props.selectedRange.to !== state.to)
+    ) {
+      return {
+        from: props.selectedRange.from,
+        to: props.selectedRange.to,
+        enteredTo: props.selectedRange.to,
+      };
+    }
+
+    return null;
+  }
+
+  handleDayClick = day => {
+    const { from, to } = this.state;
+
+    if (isSelectingFirstDay(from, to, day)) {
+      this.setState(
+        {
+          from: day,
+          to: null,
+          enteredTo: null,
+        },
+        () => this.props.onChange({ from: day, to: null }),
+      );
+    } else {
+      this.setState(
+        {
+          to: day,
+          enteredTo: day,
+        },
+        () => this.props.onChange({ from, to: day }),
+      );
+    }
+  };
+
+  handleDayMouseEnter = day => {
+    const { from, to } = this.state;
+
+    if (!isSelectingFirstDay(from, to, day)) {
+      this.setState({
+        enteredTo: day,
+      });
+    }
+  };
+
+  render() {
+    const { className, size, ...others } = this.props;
+    const { from, enteredTo } = this.state;
+    const modifiers = { from, to: enteredTo };
+    const selectedDays = [from, { from, to: enteredTo }];
+
+    const classNames = cx(
+      theme['date-picker'],
+      theme[`is-${size}`],
+      {
+        [theme['has-range']]: hasRange(selectedDays),
```

Comment:
`selectedDays` is an array, but in the implementation of `hasRange` you're doing an `Object.keys`, I'm confused 😄

Reply:
I realized that I could completely eliminate this `hasRange` function since we already know we're going to select a range. The components name therefore is `DatePickerRange`.

 :ok: by @driesd on 04/06/2018 12:20:01